src/main/java/com/vmware/vipclient/i18n/util/LocaleUtility.java

```diff
@@ -146,9 +146,12 @@ public static Locale pickupLocaleFromList(Set<Locale> locales,
                                               Locale preferredLocale) {
         Locale bestMatch = Locale.lookup(Arrays.asList(new Locale.LanguageRange(fmtToMappedLocale(preferredLocale).toLanguageTag())), locales);
 
-        // For any Chinese locale (zh-*) that is not supported, use the fallback locale even if "zh" is supported.
-        if (bestMatch != null && bestMatch.toLanguageTag().equals("zh") && !preferredLocale.toLanguageTag().equals("zh")) {
-            return null;
+        // For any Chinese locale (zh-*) that is not supported (except for zh-Hans and zh-Hant), use the fallback locale even if "zh" is supported.
+        if (preferredLocale.getLanguage().equals("zh")) {
+            Locale zhLocale = fmtToMappedLocale(preferredLocale);
+            if (!locales.contains(zhLocale) && !zhLocale.toLanguageTag().equals("zh-Hans") && !zhLocale.toLanguageTag().equals("zh-Hant")) {
+                return null;
+            }
```

Comment:
```suggestion
       if (preferredLocale.getLanguage().equals("zh")) {
            Locale zhLocale = fmtToMappedLocale(preferredLocale);
            if (!locales.contains(zhLocale)) {
                String langTag = zhLocale.toLanguageTag();
                if (langTag.equals("zh-Hans") || langTag.equals("zh-Hant")) {
                    return null;
                }
            }
        }
```

Reply:
I understand you want it to be more readable. There is incorrect logic in your suggestion though. It should be `if (!langTag.equals("zh-Hans") && !langTag.equals("zh-Hant"))` in line 160. Please review the new code.